upcloud/resource_upcloud_objectstorage.go

```diff
@@ -0,0 +1,185 @@
+package upcloud
+
+import (
+	"context"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/request"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/service"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceUpCloudObjectStorage() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceObjectStorageCreate,
+		ReadContext:   resourceObjectStorageRead,
+		UpdateContext: resourceObjectStorageUpdate,
+		DeleteContext: resourceObjectStorageDelete,
+		Schema: map[string]*schema.Schema{
+			"size": {
+				Description:  "The size of the object storage bucket in gigabytes",
```

Comment:
Remove "bucket" from all of these descriptions, e.g. here "The size of the object storage in gigabytes".

Or change "bucket" => "instance" if you prefer, though previously we made the decision to just call individual object storages as "object storages", not instances 😄 

Buckets need to be created with a separate S3 API call